src/Tools/NodeDocumentationMarkdownGenerator/Commands/FromDirectoryCommand.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NodeDocumentationMarkdownGenerator.Verbs;
+
+namespace NodeDocumentationMarkdownGenerator.Commands
+{
+    internal static class FromDirectoryCommand
+    {
+        /// <summary>
+        /// Creates markdown files using the fromdirectory verb
+        /// </summary>
+        /// <param name="opts"></param>
+        internal static void HandleDocumentationFromDirectory(FromDirectoryOptions opts)
+        {
+            Program.VerboseMode = opts.Verbose;
+            var searchOption = opts.RecursiveScan ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            var fileInfos = new List<MdFileInfo>();
+            fileInfos.AddRange(ScanFolderForAssemblies(opts.InputFolderPath, opts.Filter, opts.ReferencePaths, searchOption));
+            if (opts.IncludeCustomNodes)
+            {
+                fileInfos.AddRange(ScanFolderForCustomNodes(opts.InputFolderPath, searchOption));
+            }
+
+            MarkdownHandler.CreateMdFilesFromFileNames(fileInfos, opts.OutputFolderPath, opts.Overwrite, opts.CompressImages, opts.CompressGifs, opts.DictionaryDirectory, opts.LayoutSpecPath);
+        }
+
+        private static List<MdFileInfo> ScanFolderForCustomNodes(string inputFolderPath, SearchOption searchOption)
+        {
+            var allDyfs = Directory.GetFiles(inputFolderPath, "*.dyf", searchOption).Select(x => new FileInfo(x)).ToList();
+            var fileInfos = new List<MdFileInfo>();
+            foreach (var cn in allDyfs)
+            {
+                var fileInfo = MdFileInfo.FromCustomNode(cn.FullName);
+                if (fileInfo is null) continue;
+
+                fileInfos.Add(fileInfo);
+            }
+            return fileInfos;
+        }
+
+        private static List<MdFileInfo> ScanFolderForAssemblies(string inputFolderPath, IEnumerable<string> filter, IEnumerable<string> referencePaths, SearchOption searchOption)
+        {
+            var allAssembliesFromInputFolder = Directory.GetFiles(inputFolderPath, "*.*", searchOption)
```

Comment:
Not sure how many filles we expect to parse here.
`EnumerateFiles` might have some benefits...
`
The EnumerateFiles and GetFiles methods differ as follows: When you use EnumerateFiles, you can start enumerating the collection of names before the whole collection is returned; when you use GetFiles, you must wait for the whole array of names to be returned before you can access the array. Therefore, when you are working with many files and directories, EnumerateFiles can be more efficient.
`

Reply:
I'd rather stick with the simplicity of `GetFiles()` - this is not going to affect users, and image compression time completely dwarfs the time it takes to read these files.

I'm thinking back to the last time I ran into trouble groking deferred execution and `EnumerateFiles`.